### PR TITLE
Additional spa data from messages

### PR DESCRIPTION
**Pull request by natekspencer**
Fixed heatmode, identification of light2, timescale, and spelling of Fahrenheit

Added additional data from system information, setup parameters, module identification, and filter cycle.

### CODE REVIEW

File: pybalboa/balboa.py [hunk 1]
```diff
@@ -29,73 +29,56 @@
...
- *    Poly          = 0x07
- *    Xor_In        = 0x02
- *    ReflectIn     = False
- *    Xor_Out       = 0x02
- *    ReflectOut    = False
- *    Algorithm     = bit-by-bit
-
-https://github.com/garbled1/gnhast/blob/master/balboacoll/collector.c
-"""
-
```
Comment (garbled1) [inline pybalboa/balboa.py:89]: Don't remove the documentation for the CRC

**Review by garbled1 — CHANGES_REQUESTED** [review]
OK, yeah, I generally like this.  I need to test it out with my spa still though... please restore the docs on the CRC please.

File: pybalboa/balboa.py [hunk 2]
```diff
@@ -237,379 +241,278 @@ async def send_panel_req(self, ba, bb):
...
+        12 - number of pumps
+        13 - unknown
         """
 
-        macaddr = f'{data[8]:x}:{data[9]:x}:{data[10]:x}'\
-            f':{data[11]:x}:{data[12]:x}:{data[13]:x}'
+        self.tmin[0] = [data[7], self.to_celsius(data[7])]
+        self.tmax[0] = [data[8], self.to_celsius(data[8])]
+        self.tmin[1] = [data[9], self.to_celsius(data[9])]
+        self.tmax[1] = [data[10], self.to_celsius(data[10])]
```
Comment (garbled1) [inline pybalboa/balboa.py:464]: I'm slightly concerned about this.. we pull in tmax/tmin to celcius, but what if the spa is set in F?

**Review by garbled1 — CHANGES_REQUESTED** [review]

File: pybalboa/balboa.py (hunk 1 above)
Comment (natekspencer) [reply pybalboa/balboa.py:89]: Done! Apologies about removing that. It was my (naive) attempt at removing comments

File: pybalboa/balboa.py (hunk 2 above)
Comment (natekspencer) [reply pybalboa/balboa.py:480]: This stores both the Fahrenheit value, as well as the Celsius value of min and max temp values the spa supports. The spa, regardless of whether it is set to Fahrenheit or Celsius, always reports the value in Fahrenheit and then has to be converted to Celsius.

Comment (natekspencer) [reply pybalboa/balboa.py:483]: I added a comment around that section to clarify

File: pybalboa/balboa.py (hunk 1 above)
Comment (natekspencer) [reply pybalboa/balboa.py:89]: Also, if you'd like to try it out in home-assistant, this is my HACS repository I have setup that loads pybalboa from my github repository currently: https://github.com/natekspencer/spaclient_hacs

Once it's merged and a new version of pybalboa is pushed out to pypi, I'd love to change it to load from there!

**Comment by garbled1** [comment]
OK.. I will have time on friday to rip my spa off my prod instance and fire it onto my dev instance, and get this fully tested.  Thanks for making the fixes.

**Pull request merged**